File: Data/Entities/Ingredient.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;

namespace KitchenBear.Data.Entity;

public class Ingredient
{
    public int id { get; set; }
    public string Name { get; set; } = "";
    public string Unit { get; set; } = "Units";
    public bool Stocked { get; set; } = false;
    public bool Perishable { get; set; } = false;
    public int Calories { get; set; } = 0;
    public float Price { get; set; } = 0;
    public IEnumerable<Recipe> Recipes { get; set; } = new List<Recipe>();
}

File: ClientApp/src/app/classes/ingredient.ts
export interface IIngredient {
  id: number;
  name: string;
  unit: string;
  stocked: boolean;
  perishable: boolean;
  calories: number;
  price: number;
}
export class Ingredient implements IIngredient {
  public id: number = -1;
  public name: string = "";
  public unit: string = "Units";
  public stocked: boolean = false;
  public perishable: boolean = false;
  public calories: number = 0;
  public price: number = 0;
  // public IEnumerable<Recipe> Recipes { get; set; } = new List<Recipe>();
}
